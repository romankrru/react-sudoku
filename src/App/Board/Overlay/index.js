import React from "react";

import styles from "./index.module.css";

const Overlay = (props) => (
	<div className={styles.overlay}>
		<div onClick={props.resume} className={styles.overlayResume}></div>
	</div>
);

export default Overlay;
