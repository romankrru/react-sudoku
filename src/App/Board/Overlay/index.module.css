.overlay {
    position: absolute;
    top: 0%;
    left: 0%;

    width: 100%;
    height: 100%;

    background: rgba(255, 255, 255, 0.3);
    backdrop-filter: blur(6px);
}

.overlayResume {
    position: absolute;
    top: 50%;
    left: 50%;
    margin-left: -34px;
    margin-top: -34px;

    width: 68px;
    height: 68px;

    background-color: #5392F6;
    background-image:  url('./assets/triangle.svg');
    background-repeat: no-repeat;
    background-position: 25px center;
    border-radius: 50%;
    box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.25);
    cursor: pointer;
    transition: all .2s ease;
}

.overlayResume:hover {
    box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.25);
}
