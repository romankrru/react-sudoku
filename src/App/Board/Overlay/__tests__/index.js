import React from "react";
import { render, fireEvent } from "@testing-library/react";

import Overlay from "../";

describe("<Overlay />", () => {
	it("should render without crashes", () => {
		const view = render(<Overlay resume={jest.fn} />);
		expect(view.container).toMatchSnapshot();
	});

	it("should handle resume click", () => {
		const resumeMock = jest.fn(() => {});
		const view = render(<Overlay resume={resumeMock} />);

		expect(resumeMock).not.toHaveBeenCalled();
		fireEvent.click(view.container.querySelector(".overlayResume"));
		expect(resumeMock).toHaveBeenCalled();
		expect(resumeMock).toHaveBeenCalledTimes(1);
	});
});
