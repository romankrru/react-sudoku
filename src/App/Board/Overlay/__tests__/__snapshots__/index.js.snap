// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overlay /> should render without crashes 1`] = `
<div>
  <div
    class="overlay"
  >
    <div
      class="overlayResume"
    />
  </div>
</div>
`;
